Project/Client/Inspector.cpp

```diff
@@ -199,6 +199,21 @@ void Inspector::CheckComponent(COMPONENT_TYPE _type)
 		return;
 	}
 
+	// 두 개 이상의 렌더 컴포넌트를 추가하려고 할 시
+	if (COMPONENT_TYPE::MESHRENDER == _type || COMPONENT_TYPE::TILEMAP == _type || COMPONENT_TYPE::PARTICLESYSTEM == _type
+		|| COMPONENT_TYPE::SKYBOX == _type || COMPONENT_TYPE::DECAL == _type)
+	{
+		if (nullptr != m_TargetObject->GetComponent((COMPONENT_TYPE::MESHRENDER))
+			|| nullptr != m_TargetObject->GetComponent((COMPONENT_TYPE::TILEMAP))
+			|| nullptr != m_TargetObject->GetComponent((COMPONENT_TYPE::PARTICLESYSTEM))
+			|| nullptr != m_TargetObject->GetComponent((COMPONENT_TYPE::SKYBOX))
+			|| nullptr != m_TargetObject->GetComponent((COMPONENT_TYPE::DECAL)))
+		{
+			MessageBoxA(nullptr, "Already contains the other render component", "Can't add more than one render component!", MB_OK);
+			return;
+		}
+	}
```

Comment:
이렇게 enum비교하는게 지금보단 유지보수에 유리해보이고, 그걸 넘어서 isRendercomponent(type) 같은 함수가 있으면 좋을듯?

```suggestion
	if (COMPONENT_TYPE::MESHRENDER <= _type && _type < COMPONENT_TYPE::END )
	{
		if (nullptr != m_TargetObject->GetRenderComponent())
		{
			MessageBoxA(nullptr, "Already contains the other render component", "Can't add more than one render component!", MB_OK);
			return;
		}
	}
```

Reply:
isRenderComponent함수가 들어갈 위치?에 대한 고민이 좀 생김